### DISCUSSION TITLE
client libraries: Provide API for specifying cache dir

### DISCUSSION THREAD

**Discussion by sscarduzio** [envkey/envkey#7] [Ideas and feedback]
At the moment, I can toggle the use of cache:

```
import envkey

all_conf = envkey.fetch_env(cache_enabled=True)

```

But AFAIU, the cache files are all held under `~/.envkey/cache/`. However, my unix user is launching 4 processes at boot. All of which make use of Envkey in different ways (each have their own Envkey app, some share it). I would prefer to keep the cache files under their own directory, near each app's code, i.e. `~/app-1/.envkey/cache`.

**Comment by danenania** [comment]
It's possible to make this configurable, but I'm not sure that I see the need... if it's a security concern, the cache files are encrypted and can only be decrypted with the `ENVKEY`.

**Comment by danenania** [comment]
I should also note that if you use [envkey-source](https://docs-v2.envkey.com/docs/envkey-source) directly from the command line instead of envkey-python, you can specify the cache file location.

**Comment by yuvi3005** [comment]
Where do i find the open source API

On Tue, Apr 19, 2022 at 10:06 AM Dane Schneider ***@***.***>
wrote:

> I should also note that if you use envkey-source
> <https://docs-v2.envkey.com/docs/envkey-source> directly from the command
> line instead of envkey-python, you can specify the cache file location.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/envkey/envkey/issues/6#issuecomment-1102886820>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXCGCUTUZJ4TEMYQ5VXDPSDVF3RZPANCNFSM5TZNJEAQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Sent from my iPhone

**Comment by danenania** [comment]
@yuvi3005 Right here in this repo. https://github.com/envkey/envkey/tree/main/public/app/api
